setuptools/package_index.py

```diff
@@ -293,13 +292,13 @@ def __init__(
         self.package_pages = {}
         self.allows = re.compile('|'.join(map(translate, hosts))).match
         self.to_scan = []
-        use_ssl = (
-            verify_ssl
-            and ssl_support.is_available
-            and (ca_bundle or ssl_support.find_ca_bundle())
-        )
-        if use_ssl:
-            self.opener = ssl_support.opener_for(ca_bundle)
+        if verify_ssl or ca_bundle:
+            import ssl  # deferred import because ssl technically isn't mandatory
+
+            self.opener = partial(
+                urllib.request.urlopen,
+                context=ssl.create_default_context(cafile=ca_bundle)
+            )
         else:
             self.opener = urllib.request.urlopen
```

Comment:
it seems verify_ssl=False doesn't behave as expected. The default urllib.request.urlopen now verifies TLS correctly

```suggestion
        if ca_bundle:
            import ssl  # deferred import because ssl technically isn't mandatory

            self.opener = partial(
                urllib.request.urlopen,
                context=ssl.create_default_context(cafile=ca_bundle)
            )
        else:
            self.opener = urllib.request.urlopen
```

Reply:
Here I'm running on a fresh ubuntu docker image after `apt install python3-setuptools ca-certificates`

```
>>> import setuptools
>>> import setuptools.package_index
>>> setuptools.package_index.PackageIndex(verify_ssl=False).opener("https://google.com")
<http.client.HTTPResponse object at 0x7f0754c604c0>
>>> setuptools.package_index.PackageIndex(verify_ssl=False).opener("https://wrong.host.badssl.com/")
Traceback (most recent call last):
  File "/usr/lib/python3.8/urllib/request.py", line 1354, in do_open
    h.request(req.get_method(), req.selector, req.data, headers,
  File "/usr/lib/python3.8/http/client.py", line 1252, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib/python3.8/http/client.py", line 1298, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.8/http/client.py", line 1247, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.8/http/client.py", line 1007, in _send_output
    self.send(msg)
  File "/usr/lib/python3.8/http/client.py", line 947, in send
    self.connect()
  File "/usr/lib/python3.8/http/client.py", line 1421, in connect
    self.sock = self._context.wrap_socket(self.sock,
  File "/usr/lib/python3.8/ssl.py", line 500, in wrap_socket
    return self.sslsocket_class._create(
  File "/usr/lib/python3.8/ssl.py", line 1040, in _create
    self.do_handshake()
  File "/usr/lib/python3.8/ssl.py", line 1309, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: Hostname mismatch, certificate is not valid for 'wrong.host.badssl.com'. (_ssl.c:1131)
```